use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,221 @@
+
+## Install and setup SPDK vhost-user target
+
+### Get source code and build SPDK
+
+Following the SPDK [getting started guide](https://spdk.io/doc/getting_started.html).
+
+### Run SPDK vhost-user target
+
+First, run the SPDK `setup.sh` script to setup some hugepages for the SPDK vhost
+target application. We recommend you use a minimum of 4GiB, enough for the SPDK
+vhost target and the virtual machine.
+This will allocate 4096MiB (4GiB) of hugepages:
+
+```bash
+$ sudo HUGEMEM=4096 scripts/setup.sh
+```
+
+Then, take directory `/var/run/kata-containers/vhost-user` as Kata's vhost-user
+device directory. Make subdirectories for vhost-user sockets and device nodes:
+
+```bash
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/block/
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/block/sockets/
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/block/devices/
+```
+
+For more details, see section [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices).
+
+Next, start the SPDK vhost target application.  The following command will start
+vhost on the first CPU core with all future socket files placed in
+`/var/run/kata-containers/vhost-user/block/sockets/`:
+
+```bash
+$ sudo app/spdk_tgt/spdk_tgt -S /var/run/kata-containers/vhost-user/block/sockets/ &
+```
+
+To list all available vhost options run the following command:
+
+```bash
+$ app/spdk_tgt/spdk_tgt -h
+```
+
+Create an experimental `vhost-user-blk` device based on memory directly:
+
+- The following RPC will create a 64MB memory block device named `Malloc0`
+with 4096-byte block size:
+
+```bash
+$ sudo scripts/rpc.py bdev_malloc_create 64 4096 -b Malloc0
+```
+
+- The following RPC will create a `vhost-user-blk` device exposing `Malloc0`
+block device. The device will be accessible via
+`/var/run/kata-containers/vhost-user/block/sockets/vhostblk0`:
+
+```bash
+$ sudo scripts/rpc.py vhost_create_blk_controller vhostblk0 Malloc0
+```
+
+## Host setup for vhost-user devices
+
+Considering the OCI specification and characteristics of vhost-user device,
+Linux reserves the block major range `240-254` to map each vhost-user block
+type to a major. Also a specific directory is used for vhost-user devices.
+The base directory is a configurable value, with the default being
+`/var/run/kata-containers/vhost-user`. It can be configured by parameter
+`vhost_user_store_path` in [Kata TOML configuration file](https://github.com/kata-containers/runtime/blob/master/README.md#configuration).
+
+The reset of the path `block` is used for block device; `block/sockets` is where
+we expect vhost-user sockets to live; `block/sockets` is where simulated block
+device node for vhost-user devices to live.
+
+For example, if using the default directory `/var/run/kata-containers/vhost-user`,
+sockets for vhost-user device are under
+
+```
+/var/run/kata-containers/vhost-user/block/sockets/
```

Comment:
I would change this to `/var/run/kata-containers/vhost-user/block/sockets/` as this is not really a code snippet.